appinventor/appengine/src/com/google/appinventor/client/editor/simple/components/MockChartData.java

```diff
@@ -37,6 +46,13 @@ public void addToChart(MockChart chart) {
         labelWidget.setHeight("0");
 
         this.chart = chart;
+        this.chartModel = chart.createChartModel();
+        setDefaultData();
+
+        this.chartModel.changeColor(getPropertyValue(PROPERTY_COLOR));
+        this.chartModel.changeLabel(getPropertyValue(PROPERTY_LABEL));
```

Comment:
Should be moved elsewhere (in it's own method preferably), or an alternative approach could be used instead (method does too much at the moment)